common/models/role.js

```diff
@@ -238,14 +240,31 @@ module.exports = function(Role) {
     debug('isOwner(): %s %s userId: %s principalType: %s',
       modelClass && modelClass.modelName, modelId, userId, principalType);
 
-    // Return false if userId is missing
+    // Resolve isOwner false if userId is missing
     if (!userId) {
       process.nextTick(function() {
         callback(null, false);
       });
       return callback.promise;
     }
 
+    // At this stage, principalType is valid in one of 2 following condition:
+    // 1. the app has a single user model and principalType is 'USER'
+    // 2. the app has multiple user models and principalType is not 'USER'
```

Comment:
I agree it doesn't make sense to call `isOwner` passing `APP` or `ROLE` as a principal type.

However, are we sure that no code path can lead to such call? Is there any safe-guard check in `isOwner` to ensure we still return a sensible result in such case?